iceoryx_hoofs/cmake/IceoryxPackageHelper.cmake

```diff
@@ -379,4 +379,7 @@ Macro(iox_add_library)
             INCLUDE_DIRECTORY include/
         )
     endif()
+
+    unset(IOX_NO_PACKAGE_SETUP)
+    unset(IOX_NO_FIND_PACKAGE_SUPPORT)
```

Comment:
The problem is this variables. Since they are defined within a macro they are also visible after the macro call. For now only these two variables are unset but I will create an issue to check which need to be unset as well or if we can create a cmake function for this

Reply:
done with #1532